vcx/libvcx/src/api/vcx.rs

```diff
@@ -147,7 +148,18 @@ fn _finish_init(command_handle: CommandHandle, cb: extern fn(xcommand_handle: Co
                                   storage_config.as_ref().map(String::as_str), storage_creds.as_ref().map(String::as_str)) {
             Ok(_) => debug!("Init Wallet Successful"),
             Err(e) => {
-                error!("Init Wallet Error {}.", e);
+                error!("Init Wallet Error: {}", e);
+                cb(command_handle, e.into());
+            }
+        };
+
+        match agent_utils::create_pub_agent() {
```

Comment:
The creation of public agent should be exposed in API as an individual function for that purpose.